### PR TITLE
Feat/backend/update rating api

### PR DESCRIPTION
**Pull request by markjoshuacarmen**
## Ticket/s

- [Update rating API](https://trello.com/c/g6RpVVau/44-api-update-product-rating)

## Changes

- add update-rating-schema
- add update-rating services

## Screenshot
image

image

## How to test?

- pnpm run dev
- open postman
- add products to Cart - https://github.com/TreshAnn/POC-EyCommerce/pull/84
- create Order from products in Cart - https://github.com/TreshAnn/POC-EyCommerce/pull/97
- create order with different status: "processing", "shipped", "delivered"
- create product rating https://github.com/TreshAnn/POC-EyCommerce/pull/106
- 
- get all user rating:
`curl --location 'localhost:3001/api/rating/get-user-ratings' \
--header 'Authorization: Bearer eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJ1c2VybmFtZSI6ImNvbnN1bWVyIiwic3ViIjoiNjUwMDAwMTIxZTJlMWJhNGU4MTM2ZGUxIiwiaWF0IjoxNjk0OTk5ODQ4LCJleHAiOjE2OTUwODYyNDh9.4z6O6f9bMOxngoVJfZSDqKyLJpnZHJrwbvhJ9IEsNF4'`

- copy the rating_Id
`curl --location --globoff --request PUT 'localhost:3001/api/rating/update/[place rating_id here]' \
--header 'Content-Type: application/json' \
--header 'Authorization: Bearer eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJ1c2VybmFtZSI6ImNvbnN1bWVyIiwic3ViIjoiNjUwMDAwMTIxZTJlMWJhNGU4MTM2ZGUxIiwiaWF0IjoxNjk0OTk5ODQ4LCJleHAiOjE2OTUwODYyNDh9.4z6O6f9bMOxngoVJfZSDqKyLJpnZHJrwbvhJ9IEsNF4' \
--data '{
    "rating" : 4,
    "title": " test ",
    "description": " "
}'`

### CODE REVIEW

File: apps/api/src/rating/rating.service.ts [hunk 1]
```diff
@@ -0,0 +1,127 @@
...
+
+    if (updatedRatingDto.rating !== undefined) {
+      existingRating.rating = updatedRatingDto.rating;
+    }
+    if (updatedRatingDto.title !== undefined) {
+      existingRating.title = updatedRatingDto.title;
+    }
+    if (updatedRatingDto.description !== undefined) {
+      existingRating.description = updatedRatingDto.description;
+    }
```
Comment (TreshAnn) [inline apps/api/src/rating/rating.service.ts]: After implementing these changes, I suggest you thoroughly test the updateRating function to ensure it functions as expected. As for the time, consider converting updateTime to Philippine Standard Time (PST). not sure if this works, but you can refer to this Stack Overflow post: https://stackoverflow.com/questions/35672248/how-to-change-date-timezone-in-mongoose

```suggestion

    const updateTime = new Date();
    updateTime.setTime(updateTime.getTime() - 2 * 60 * 1000); // 2 minutes for testing only
    //for 5 days updateTime.setDate(updateTime.getDate() - 5);

    if (existingRating.publishedDate < updateTime) {
      throw new BadRequestException(
        'You can only update your review within 2 minutes of the original review',
      );
    }

    if (updatedRatingDto.rating !== undefined) {
      existingRating.rating = updatedRatingDto.rating;
    }
    if (updatedRatingDto.title !== undefined) {
      existingRating.title = updatedRatingDto.title;
    }
    if (updatedRatingDto.description !== undefined) {
      existingRating.description = updatedRatingDto.description;
    }

    existingRating.reviewDate = updateTime;
```

**Review by TreshAnn — CHANGES_REQUESTED** [review]

**Review by DanDalanDone69 — COMMENTED** [review]
Use dayjs, don’t use moment

File: pnpm-lock.yaml
Comment (brentmillen) [inline pnpm-lock.yaml:1]: remove imported library

File: apps/api/src/rating/rating.service.ts [hunk 2]
```diff
@@ -33,28 +31,19 @@ export class RatingService {
     createRatingDto: CreateRatingDto,
   ): Promise<Rating> {
     const userId = req._id;
-
     const authUser = await this.authService.findOne(userId);
-
-    // flag to check if the item is found
     let itemFound = false;
```
Comment (brentmillen) [inline apps/api/src/rating/rating.service.ts:47]: just use an if else state for this

**Review by brentmillen — COMMENTED** [review]
see comments

**Review by TreshAnn — APPROVED** [review]

File: pnpm-lock.yaml
Comment (markjoshuacarmen) [reply pnpm-lock.yaml:1]: this is noted thanks

**Pull request merged**